modules/basics/list/list.h

```diff
@@ -0,0 +1,283 @@
+#ifndef LIST_H
+#define LIST_H
+
+struct list_head {
+  struct list_head *prev, *next;
+};
+
+/*
+ * It would be nice to make these constants architecture dependant as Linux does
+ * in https://github.com/torvalds/linux/blob/master/include/linux/poison.h
+ */
+#define LIST_POISON1 ((void *)0x00100100)
+#define LIST_POISON2 ((void *)0x00200200)
+
+#define container_of(ptr, type, member)                                        \
+  ({                                                                           \
+    const typeof(((type *)0)->member) *__mptr = (ptr);                         \
+    (type *)((char *)__mptr - offsetof(type, member));                         \
+  })
+
+#define LIST_HEAD_INIT(name)                                                   \
+  { &(name), &(name) }
+
+/**
+ * LIST_HEAD - Declare a list head and initialize it
+ * @name: name
+ */
+#define LIST_HEAD(name) struct list_head name = LIST_HEAD_INIT(name)
```

Comment:
Are this macros ever used? Or are they just here to complete the linked list API?